### PR TITLE
v1.0.0 blog

### PR DESCRIPTION
**Pull request by vwxyzjn**
## Description
v1.0.0 blog

## Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] New algorithm
- [x] Documentation

### CODE REVIEW

File: docs/blog/posts/cleanrl-v1.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+
+<!-- more -->
+
+## Reworked documentation
+
+One of the biggest change of the v1 release is the added documentation at [docs.cleanrl.dev](https://docs.cleanrl.dev). Having great documentation is important for building a reliable and reproducible project. We have reworked the documentation to make it easier to understand and use. For each implemented algorithm, we have documented as much as we can to promote transparency:
+
+* [Short description of the algorithm and references](/rl-algorithms/ppo/#overview)
+* [A list of implemented varaint](/rl-algorithms/ppo/#implemented-variants)
```
Comment (joaogui1) [inline docs/blog/posts/cleanrl-v1.md:25]: varaint -> variant

File: docs/blog/posts/cleanrl-v1.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+* [The usage information](/rl-algorithms/ppo/#usage)
+* [The explanation of the logged metrics](/rl-algorithms/ppo/#explanation-of-the-logged-metrics)
+* [The documentation of implementation details](/rl-algorithms/ppo/#implementation-details)
+* [Experimental results](/rl-algorithms/ppo/#experiment-results)
+
+Here is a list of the algorithm varaints and their documentation:
```
Comment (joaogui1) [inline docs/blog/posts/cleanrl-v1.md:31]: varaints -> variant

File: docs/blog/posts/cleanrl-v1.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+categories:
+  - Blog
+---
+
+
+
+# CleanRL v1 Release
+
+
+🎉 We are thrilled to announce the v1.0.0 CleanRL Release. CleanRL has come a long way making high-quality deep reinforcement learning implementations easy to understand and reproducible. Along with our [CleanRL paper's recent publication in Journal of Machine Learning Research](https://www.jmlr.org/papers/v23/21-1342.html), our v1.0.0 release includes reworked documentation, new algorithm variants, support for JAX, hyperparameter tuning utilties, and more. This release is a major milestone for the project and we are excited to share it with you. Over 90 PRs were merged to make this release possible. We would like to thank all the contributors who made this release possible.
```
Comment (joaogui1) [inline docs/blog/posts/cleanrl-v1.md:15]: Maybe link to the JAX github?

File: docs/blog/posts/cleanrl-v1.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+    --output-filename compare.png \
+    --scan-history \
+    --metric-last-n-average-window 100 \
+    --report
+```
+
+which generates the following image
+
+<img max-width="500px" src="/static/blog/cleanrl-v1/rlops.png">
+
```
Comment (kinalmehta) [inline docs/blog/posts/cleanrl-v1.md:248]: Do you plan on merging RLops before 1.0 release?
This is coming off as it is released in 1.0
Maybe a little rewording would be good if that is not the case.

**Review by kinalmehta — COMMENTED** [review]
Everything else looks great!

File: docs/blog/posts/cleanrl-v1.md (hunk 4 above)
Comment (vwxyzjn) [reply docs/blog/posts/cleanrl-v1.md]: Do not plan to merge it before 1.0 release. I changed the phrasing to “we are building an RLops utility”

**Review by kinalmehta — APPROVED** [review]

**Review by dosssman — APPROVED** [review]
Look good over here.
Great work.

**Review by dipamc — APPROVED** [review]
Looks great. Excited about the RLops tool. Maybe highlight some speedup numbers for JAX variants in the blog.

**Comment by dipamc** [comment]
As a fun addition it might be a good moment for CleanRL to get a logo, how about some ASCII art?

**Comment by joaogui1** [comment]
LGTM!

**Review by joaogui1 — APPROVED** [review]

**Review by sdpkjc — APPROVED** [review]

File: docs/blog/posts/cleanrl-v1.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
+---
+date: 2022-10-05
+authors: [costa]
+description: >
+  🎉 I am thrilled to announce the v1.0.0b2 CleanRL Beta Release. This new release comes with exciting new features. First, we now support JAX-based learning algorithms, which are usually faster than the `torch` equivalent! Here are the docs of the new JAX-based DQN, TD3, and DDPG implementations:
```
Comment (yooceii) [inline docs/blog/posts/cleanrl-v1.md:5]: v1.0.0b2 CleanRL Beta Release or v1.0.0 CleanRL Release?

File: docs/blog/posts/cleanrl-v1.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+
+## New algorithm variants, support for JAX
+
+We now support JAX-based learning algorithm variants, which are usually faster than the `torch` equivalent! Here are the docs of the new JAX-based DQN, TD3, and DDPG implementations:
+
+
+* [`dqn_atari_jax.py`](https://docs.cleanrl.dev/rl-algorithms/dqn/#dqn_atari_jaxpy) [@kinalmehta](https://github.com/kinalmehta) in [:material-github: vwxyzjn/cleanrl#222](https://github.com/vwxyzjn/cleanrl/pull/222)
+    * about 25% faster than `dqn_atari.py`.
+* [`dqn_jax.py`](https://docs.cleanrl.dev/rl-algorithms/dqn/#dqn_jaxpy) [@kinalmehta](https://github.com/kinalmehta) in [:material-github: vwxyzjn/cleanrl#222](https://github.com/vwxyzjn/cleanrl/pull/222)
+    * about 4% faster than `dqn.py`.
```
Comment (yooceii) [inline docs/blog/posts/cleanrl-v1.md:74]: I thought it was 40%?

File: docs/blog/posts/cleanrl-v1.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+
+
+## Tooling improvements
+
+We love tools! The v1.0.0 release comes with a series of DevOps improvements, including pre-commit utilities, CI integration with GitHub to run end-to-end test cases. We also make available a new hyperparameter tuning tool and a new tool for running benchmark experiments.
+
+### DevOps
+
+We added a pre-commit utility to help contributors to format their code, check for spelling, and removing unused variables and imports before submitting a pull request (see [Contribution guide](http://127.0.0.1:8000/contribution/#pre-commit-utilities) for more detail).
```
Comment (yooceii) [inline docs/blog/posts/cleanrl-v1.md:111]: Why is it a local address

**Review by yooceii — DISMISSED** [review]
Overall LGTM

File: docs/blog/posts/cleanrl-v1.md (hunk 6 above)
Comment (vwxyzjn) [reply docs/blog/posts/cleanrl-v1.md:74]: Not according to https://docs.cleanrl.dev/rl-algorithms/dqn/#dqn_jaxpy. However, note that the `dqn.py` uses CPU and `dqn_jax.py` uses GPU and the video recording could be computationally expensive.

Comment (kinalmehta) [reply docs/blog/posts/cleanrl-v1.md:74]: I might have missed updating the proper performance metric here. BTW on the link you provided, it's mentioned 4x, in the blog it's mentioned 4% and actually is around 20-25% (wandb timing plots).
I'll create a new pull request to fix all these.

Also, I didn't know the PyTorch version was benchmarked using CPU. Based on my testing, the JAX version on the CPU is approx 2x faster relative to PyTorch.

Comment (vwxyzjn) [reply docs/blog/posts/cleanrl-v1.md:74]: Makes sense. Let me omit `dqn_jax.py` from the highlight in the announcement for now.
